includes/abstracts/abstract-wc-product.php

```diff
@@ -1163,6 +1388,48 @@ protected function update_post_meta() {
 	protected function update_terms() {
 		wp_set_post_terms( $this->get_id(), $this->data['category_ids'], 'product_cat', false );
 		wp_set_post_terms( $this->get_id(), $this->data['tag_ids'], 'product_tag', false );
+		wp_set_post_terms( $this->get_id(), array( $this->data['shipping_class_id'] ), 'product_shipping_class', false );
+	}
+
+	/**
+	 * Update attributes which are a mix of terms and meta data.
+	 *
+	 * @since 2.7.0
+	 */
+	protected function update_attributes() {
+		$attributes  = $this->get_attributes();
+		$meta_values = array();
+
+		if ( $attributes ) {
+			foreach ( $attributes as $attribute_key => $attribute ) {
+				$value = '';
+
+				if ( is_null( $attribute ) ) {
+					if ( taxonomy_exists( $attribute_key ) ) {
+						// Handle attributes that have been unset.
+						wp_set_object_terms( $this->get_id(), array(), $attribute_key );
+					}
+					continue;
+
+				} elseif ( $attribute->is_taxonomy() ) {
+					wp_set_object_terms( $this->get_id(), wp_list_pluck( $attribute->get_terms(), 'term_id' ), $attribute->get_name() );
+
+				} else {
+					$value = implode( ' ' . WC_DELIMITER . ' ', $attribute->get_options() );
```

Comment:
I notice quite a few uses of imploding with WC_DELIMITER | -- it might be useful to have an additional helper function to get options a string.